controller/controller.go

```diff
@@ -26,33 +26,31 @@ type Controller interface {
 }
 
 type controller struct {
-	updater       Updater
-	client        k8s.Client
-	watcher       k8s.Watcher
-	started       bool
-	startStopLock sync.Mutex
-	serviceDomain string
+	updater     Updater
+	client      k8s.Client
+	watcher     k8s.Watcher
+	watcherDone sync.WaitGroup
```

Comment:
As we only ever add 1 to this (I think) we could use a condition variable. WDYT?